Sming/SmingCore/Data/Stream/Base64OutputStream.cpp

```diff
@@ -25,7 +25,6 @@ int Base64OutputStream::encode(uint8_t* source, size_t sourceLength, uint8_t* ta
 	int count = 0;
 	if(sourceLength == 0) {
 		count = base64_encode_blockend((char*)target, &state);
-		count--; // the last byte is a newline. we don't need it.
```

Comment:
Is the suggested change tested? A newline at the end of the block is probably not something that we need.

Reply:
This compliments a change I made to base64_encode_blockend, removing the trailing newline character.

I haven't tested the changes to Base64OutputStream, but it should be clear from inspection that the two changes cancel each other out.

Where base64 encoding is used for smtp/websocket authentication there must absolutely be no superfluous newline characters; this is why I elected to change base64_encode_blockend (as above). This is OK provided the output line length doesn't exceed 72 characters; I use websockets (as a server) extensively so yes discovered this empirically.

In light of this I've made another small change to encoding so that 'chars_per_line' must be specified in base64_init_encodestate. That way we can be certain line breaks are handled exactly as required. The code in base64.cpp specifies chars_per_line = 0 to disable line breaking completely.

Code comments updated to clarify this.

Reply:
Perfect and thanks for the clarification.